src/main/kotlin/com/markskelton/settings/ThemeSettingsUI.kt

```diff
@@ -0,0 +1,132 @@
+import com.markskelton.settings.ThemeSettings.Companion.constructSettingModel
+import java.net.URI
+import javax.swing.JComponent
+
+data class ThemeSettingsModel(
+  var isBold: Boolean,
+  var isVivid: Boolean,
+  var isItalic: Boolean
+)
+
+class ThemeSettingsUI : DumbAware, SearchableConfigurable {
+
+  companion object {
+    const val THEME_SETTINGS_DISPLAY_NAME = "One Dark Theme Settings"
+    private const val REPOSITORY = "https://github.com/one-dark/jetbrains-one-dark-theme"
+    private val CHANGELOG_URI = URI("$REPOSITORY/blob/master/CHANGELOG.md")
+    private val ISSUES_URI = URI("$REPOSITORY/issues")
+    private val MARKETPLACE_URI = URI("https://plugins.jetbrains.com/plugin/11938-one-dark-theme")
+  }
+
+  override fun getId(): String = "com.markskelton.ThemeSettings"
+
+  override fun getDisplayName(): String =
+    THEME_SETTINGS_DISPLAY_NAME
+
+  private val initialThemeSettingsModel = constructSettingModel()
+
+  private val themeSettingsModel = initialThemeSettingsModel.copy()
+
+  override fun isModified(): Boolean {
+    return initialThemeSettingsModel != themeSettingsModel
+  }
+
+  override fun apply() {
+    persistChanges()
+    ApplicationManager.getApplication().messageBus.syncPublisher(
+      THEME_CONFIG_TOPIC
+    ).themeConfigUpdated(ThemeSettings.instance)
+  }
+
+  private fun persistChanges() {
+    registerSettingsChange(themeSettingsModel.isBold, {
+      ThemeSettings.instance.isBold
+    }) {
+      ThemeSettings.instance.isBold = it
+    }
+    registerSettingsChange(themeSettingsModel.isVivid, {
+      ThemeSettings.instance.isVivid
+    }) {
+      ThemeSettings.instance.isVivid = it
+    }
+    registerSettingsChange(themeSettingsModel.isItalic, {
+      ThemeSettings.instance.isItalic
+    }) {
+      ThemeSettings.instance.isItalic = it
+    }
+  }
+
+  override fun createComponent(): JComponent? =
+    createSettingsPane()
+
+  private fun createSettingsPane(): DialogPanel =
+    panel {
+      titledRow("Main Settings") {
+//        row {
+//          cell {
+//            checkBox(
+//              "Bold Characters",
+//              themeSettingsModel.isBold,
+//              comment = "Uses bold fonts for certain language keywords",
+//              actionListener = { _, component ->
+//                themeSettingsModel.isBold = component.isSelected
+//              }
+//            )
+//          }
+//        }
+        row {
+          cell {
+            checkBox(
+              "Italic Characters",
+              themeSettingsModel.isItalic,
+              comment = "Uses italic font for language keywords and comments",
+              actionListener = { _, component ->
+                themeSettingsModel.isItalic = component.isSelected
+              }
+            )
+          }
+        }
+        row {
+          cell {
+            checkBox(
+              "Vivid Pallette",
+              themeSettingsModel.isVivid,
+              comment = "Uses the One-Dark vivid color pallette",
```

Comment:
```suggestion
              comment = "Uses the One Dark vivid color palette",
```

Reply:
Lol, since there are two `t`'s I guess I needed to balance it out with `L`'s :joy: